### ISSUE TITLE
Integrate convex-auth

### ISSUE THREAD

**Issue by shiftlabs1** [get-convex/convex-svelte#11]
It would be nice to get convex-auth integrated here , providing a compatible module that will enable client side and server side Auth in sveltekit project. convex-auth does not have a plain JS/TS lib as with everything convex, it only react implementation which is a challenge to use especially when one does not use react

**Comment by thomasballinger** [comment]
Absolutely, the first order of business here will probably be just an example of any kind of auth, likely with Clerk first. That should be a smaller project, just adding it to the example in this repo.

Then getting it working with Convex Auth will be next.

**Comment by Wesley7104** [comment]
Yes this would be amazing!!

**Comment by jasongitmail** [comment]
I just implemented Github OAuth with Convex on SvelteKit and React projects side by side.

The Svelte Kit implementation was blocked on lacking Svelte versions for these methods from the React lib, particularly the top two:
```ts
import { useAuthActions } from "@convex-dev/auth/react";
import { Authenticated, Unauthenticated, AuthLoading } from "convex/react";

import { useAuthToken } from "@convex-dev/auth/react";
```

But instead of the "provider style wrapper" components in the 2nd LOC, it'd be Svelte-like to use booleans of `isAuthenticated` and `isAuthLoading` booleans and let the dev use templating conditionals. (Tbh, I'd prefer booleans on the React side too, but that works at least.)

**Comment by o-namacode** [comment]
Is auth fixed yet?? Any resources to help? 

I'm trying to get basic convex password working, however auth isn't reflected on the client side queries; the same queries on server work.

I'd much rather not have to add another dependency to my project just to get around this

**Comment by mmailaender** [comment]
@o-namacode 

Convex Auth isn’t build in into convex-svelte itself — mainly because there isn’t a single auth approach in the Convex ecosystem (Convex Auth, Better Auth, Clerk, WorkOS, etc.), so the core client stays unopinionated.

The issue you’re running into (auth not reflected in client-side queries) is exactly the missing Svelte auth layer.
You can implement that yourself — but realistically that means rebuilding the same glue code (auth state, loading handling, subscriptions, etc.), which is quite a bit of work for something that’s already solved.

So in practice you have two solid options:
1. Use convex-auth-svelte
→ Minimal addition, gives you a proper Svelte integration for Convex Auth.

2. Use the maintained Svelte client fork
→ [@mmailaender/convex-svelte](https://github.com/mmailaender/convex-svelte)
→ Provides a generic useAuth (similar to React) with isAuthenticated / isLoading.

If you really want to avoid adding a dependency, I’d strongly recommend at least copying the implementation from one of these instead of starting from scratch — otherwise you’ll likely run into the same edge cases.

**Comment by mmailaender** [comment]
As convex-svelte is unopinionated about the auth provider I'm closing this issue. If you use Convex Auth or Convex Better Auth, you can use the following extensions for convex-svelte: 

Convex Auth: https://github.com/mmailaender/convex-auth-svelte
Convex Better Auth: https://github.com/mmailaender/convex-better-auth-svelte
